test/unit/org/apache/cassandra/index/sai/cql/QueryTimeoutTest.java

```diff
@@ -130,4 +134,26 @@ public void delayDuringTokenLookupShouldProvokeTimeoutInRangeIterator() throws T
 
         waitForEquals(queryCountName, queryTimeoutsName);
     }
+
+    @Test
+    public void abortedOperationExceptionShouldProvokeQueryFailureWithTimeoutCode() throws Throwable
+    {
+        Injection abortedOperationException = Injections.newCustom("throw_aborted_operation_exception")
+                .add(newInvokePoint()
+                        .onClass(StorageAttachedIndexSearcher.class)
+                        .onMethod("search")
+                        .atEntry())
+                .add(ActionBuilder.newActionBuilder()
+                        .actions()
+                        .doAction("throw new org.apache.cassandra.index.sai.utils.AbortedOperationException();"))
+                .build();
+        Injections.inject(abortedOperationException);
+
+        assertThatThrownBy(() -> executeNet("SELECT * FROM %s WHERE v2 = '1'"))
```

Comment:
Just out of curiosity, will this also cover queries like `SELECT <whatever> FROM %s ORDER BY <vector_column> ANN OF ? LIMIT ?` type queries too?

Reply:
Yes, it will.